lib/ansible/modules/windows/setup.ps1

```diff
@@ -349,7 +349,7 @@ if($gather_subset.Contains('processor')) {
     }
 
     $cpu_list = @( )
-    for ($i=1; $i -le ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors); $i++) {
+    for ($i=1; $i -le $win32_cs.NumberOfProcessors; $i++) {
```

Comment:
So I'm comparing the the facts from the POSIX side and `ansible_processor` is returning a fun list of processors in each thread. This change here seems like you are wanting to set this value to actual physical processors and not cores or threads. This is a change that I'm not sure we actually want.

Reply:
Looks like POSIX setup returns a list for each virtual cpu in a processor - for example, a box with two of [these processors](https://www.intel.com/content/www/us/en/products/processors/xeon/e5-processors/e5-2650-v3.html) returns (copied from AWX) :

 ansible_processor:
    '0'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '1'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '2'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '3'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '4'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '5'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '6'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '7'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '8'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '9'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '10'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '11'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '12'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '13'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '14'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '15'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '16'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '17'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '18'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '19'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '20'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '21'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '22'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '23'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '24'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '25'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '26'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '27'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '28'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '29'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '30'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '31'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '32'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '33'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '34'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '35'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '36'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '37'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '38'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
    '39'
    GenuineIntel
    Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz
 ansible_processor_cores: 10
 ansible_processor_count: 2
 ansible_processor_threads_per_core: 2
 ansible_processor_vcpus: 40
 
I pushed a commit to mirror this behavior, but perhaps it might be best to not set this as a list and make it single valued for both architectures.  I do not think there will be a time where the values would not be identical.

Reply:
MAybe add ` Select-Object -Unique` in line 358 then

Reply:
That seems reasonable to me, but it would differ from the way results are returned for a linux box with the same configuration.  I would assume that we would want parity?

Reply:
So, is the philosophy that Windows and *nix setup should return facts in the same format?

Reply:
@jborean93 maybe you can weigh in on how setup should behave.  I'm suggesting that we return results in the same format...

Reply:
The Python setup isn't that great but we should be following with what they have there.